cwrstatus/datastore.py

```diff
@@ -2,151 +2,25 @@
-
-        See self.get() parameters for more info.
-        """
-        filter = ({'bundle_name': bundle_name})
-        return self.get(filter=filter, limit=limit, skip=skip)
-
-    def update(self, _id, doc):
-        """Update a document in the collection.
-
-         If the document doesn't exist, it will be created. If it exists, it
-         will be updated.
-        :param _id: Document id.
-        :param doc: Document data.
-        """
-        doc['_updated_on'] = get_current_utc_time()
-        return self.collection.replace_one(_id, doc, upsert=True)
-
-    def distinct(self, key='bundle_name', limit=None, skip=0):
-        """Get a list of distinct values for key among all documents."""
-        limit = limit or self.limit
-        # todo: I don't like calling aggregate twice but I haven't found a way
-        # to count total number of distinct items.
-        distinct_col = self.collection.aggregate(
-            [{'$group': {'_id': '${}'.format(key), "count": {"$sum": 1}}}])
-        count = len(list(distinct_col))
-        return self.collection.aggregate(
-            [{'$group': {'_id': '${}'.format(key), "count": {"$sum": 1}}},
-             {'$skip': skip},
-             {'$limit': limit},
-             {"$sort": {"_id": 1}}]), count
-
-    def get_test_ids(self, key='test_id', limit=None, skip=0,
-                     latest_first=True, bundle=None, date=None):
-        limit = limit or self.limit
-        sort_order = pymongo.DESCENDING
-        if latest_first is False:
-            sort_order = pymongo.ASCENDING
-        match = self._generate_match_filter(bundle, date)
-        if match is not None:
-            distinct_test_id = self.collection.aggregate([
-                {"$match": match},
-                {'$group': {
-                    '_id': '${}'.format(key),
-                    'date': {"$first": "$date"},
-                }},
-                {"$sort": {"date": sort_order}},
-                {'$skip': skip},
-                {'$limit': limit},
-            ])
-        else:
-            distinct_test_id = self.collection.aggregate([
-                {'$group': {
-                    '_id': '${}'.format(key),
-                    'date': {"$first": "$date"},
-                }},
-                {"$sort": {"date": sort_order}},
-                {'$skip': skip},
-                {'$limit': limit},
-            ])
-        return distinct_test_id
-
-    @staticmethod
-    def _generate_match_filter(bundle=None, date=None):
-        match = {}
-        if bundle is not None and date is not None:
-            match = {
-                "$and": [
-                    {"bundle_name": bundle},
-                    {"date": {"$lte": date}}
-                ]
-            }
-        elif bundle is not None:
-            match = {"bundle_name": bundle}
-        elif date is not None:
-            match = {"date": {"$lte": date}}
-        return match
-
-
 class S3:
```

Comment:
This might not be needed at all. As mentioned, the data is accessible through http://data.vapour.ws, example: http://data.vapour.ws/cwr/results/index.html